package/MDAnalysis/analysis/density.py

```diff
@@ -228,7 +228,10 @@ class DensityAnalysis(AnalysisBase):
     around 4.0 (protein and not name H*)", i.e., the water oxygen atoms that
     are within 4 Å of the protein heavy atoms) then create an
     :class:`~MDAnalysis.core.groups.UpdatingAtomGroup` (see Examples).
-
+    
+    If the 'AtomGroup' instance does not contain any selection of atoms and
```

Comment:
Considering the original issue, does this reflect what should be documented here?

It's not wrong to tell users to make sure that grids are static and therefore should fit a selection over a trajectory (from a quick glance over the docstring, this isn't included, so it's not wrong to include it), but that's a separate point from the one being made about empty atomgroups.

Based on the changes done in this pull request, what would you tell a new user looking to do a density calculation with an starting empty atomgroup to do? I.e. what parts of the code aren't going to work, and which will?

Reply:
If the 'AtomGroup' instance does not contain any selection of atoms and
user defined box limits are provided, density is prepared in expectation 
for atoms in future frames in the trajectory.

Reply:
The user need to understand that without any atoms, they just get a grid with atoms to be added later on. Hopefully that's what the above message says

Reply:
Yes, but I think we need to be more informative.

If we break it down, the message should really contain the following:
1) Sentence that defines an empty :class:`UpdatingAtomGroup` in this case (i.e. no atoms in selection at the current frame during class construction).
2) What happens if you pass a :class:`UpdatingAtomGroup` which is "empty"  under default conditions? That is to say, not a user-defined grid. Why does this happen?
3) How do you circumvent this issue? i.e. using a user-defined grid including any additional considerations about  doing this.

Reply:
I think going back to the original issue clarifies this: 
1+2)  For an atomselection that is initially empty but which becomes filled over the course of the trajectory. DensityAnalysis will fail if the selection is initially empty despite using updating=True:

3) A quick solution to the above problem can be to provide a user-defined grid. However, it should capture all the atoms in future frames.